Helm: more explicit migration instructions about zone awareness (#4190)
grafana/mimir: docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-from-single-zone-with-helm.md

## Prerequisite

Depending on what version of the `mimir-distributed` Helm chart is installed currently, make sure to meet the following requirements.

> 1. In case the current version of the `mimir-distributed` Helm chart is less than 4.0.0 (version < 4.0.0).

Reviewer: ```suggestion
1. If the current version of the `mimir-distributed` Helm chart is lower than 4.0.0 (version < 4.0.0).
```

   1. Follow the upgrade instructions for 4.0.0 in the [CHANGELOG.md](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/CHANGELOG.md#400).
      In particular make sure to disable zone awareness before upgrading the chart:

      ```yaml
      ingester:
        zoneAwareReplication:
          enabled: false
      store_gateway:
        zoneAwareReplication:
          enabled: false
      rollout_operator:
        enabled: false
      ```

      > **Note**: a direct upgrade from non-zone aware ingesters to zone-aware ingesters will cause data loss othewise.

   1. If you have modified the `mimir.config` value, make sure to merge in the latest version from the chart. Or consider using `mimir.structuredConfig` instead, see [Manage the configuration of Grafana Mimir with Helm]({{< relref "../run-production-environment-with-helm/configuration-with-helm/" >}})

> 1. In case the current version of the `mimir-distributed` Helm chart is greater than 4.0.0 (version >= 4.0.0).

Reviewer: ```suggestion
- If the current version of the `mimir-distributed` Helm chart is higher than 4.0.0 (version >= 4.0.0).
```

>    1. Make sure that zone-aware replication is turned off for the component in question, e.g. for store-gateway:

Reviewer: ```suggestion
   1. Make sure that zone-aware replication is turned off for the component in question.
   
    For example, the store-gateway:
```

      ```yaml
      store_gateway:
        zoneAwareReplication:
          enabled: false
      ```

>    1. If you have modified the `mimir.config` value, make sure to merge in the latest version from the chart. Or consider using `mimir.structuredConfig` instead, see [Manage the configuration of Grafana Mimir with Helm]({{< relref "../run-production-environment-with-helm/configuration-with-helm/" >}})

Reviewer: ```suggestion
   1. If you have modified the `mimir.config` value, either make sure to merge in the latest version from the chart, or consider using `mimir.structuredConfig` instead.
   
    For more information, see [Manage the configuration of Grafana Mimir with Helm]({{< relref "../run-production-environment-with-helm/configuration-with-helm/" >}}).
```
